django/forms/widgets.py

```diff
@@ -268,6 +268,7 @@ class Input(Widget):
 
     def __init__(self, attrs=None):
         if attrs is not None:
+            attrs = attrs.copy()
             self.input_type = attrs.pop('type', self.input_type)
```

Comment:
Oh, ignore me. I see your change will effectively avoid pass-by-reference here which is also useful to avoid changes to `attrs` outside of `Input.__init__()` after instantiation from changing the attributes of the widget. Perhaps the following test should also be added:

```python
def test_attrs_passed_by_value(self):
    attrs = {'type': 'date'}
    widget = TextInput(attrs)
    self.check_html(widget, 'name', 'value', strict=True, html=(
        '<input type="date" name="name" value="value" />'
    ))
    attrs['type'] = 'number'  # shouldn't cause type of widget to become 'number'
    self.check_html(widget, 'name', 'value', strict=True, html=(
        '<input type="date" name="name" value="value" />'
    ))
```